optaplanner-website-root/content/download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -1656,3 +1656,137 @@ Even though OptaPlanner 8.x has always required JDK 11, this only becomes a hard
 === Rename optaplanner.solver.solve-length to optaplanner.solver.solve.duration
 
 The Micrometer meter `optaplanner.solver.solve-length` has been renamed to `optaplanner.solver.solve.duration` to follow Micrometer naming conventions and make it appear better in a variety of monitoring systems. References to `optaplanner.solver.solve-length.+++*+++` in the monitoring system should be changed to `optaplanner.solver.solve.duration.+++*+++`.
+
+== From 8.13.0.Final to 8.14.0.Final
+
+[.upgrade-recipe-major]
+=== Constraint Streams: use `forEach(...)` instead of `from(...)`
+
+We have deprecated the following methods on the `ConstraintFactory` interface:
+
+* `from(...)` in favor of `forEach(...)`,
+* `fromUnfiltered(...)` in favor of `forEachIncludingNullVars(...)`,
+* `fromUniquePair(...)` in favor of `forEachUniquePair(...)`.
+
+The new methods behave differently when it comes to nullable planning variables.
+The deprecated methods will eventually be removed in OptaPlanner 9.
```

Comment:
Messaging wise, it's not a good idea to mention "OptaPlanner 9" in a way some users could think that it's already planned for a target date (which it is not). Because when users start thinking it's coming soon (which it is not, work hasn't started on it and won't start for years), it creates fear, uncertainty and doubt for the current major version (this is real, however stupid it sounds) and they tend to go with other solutions than OptaPlanner.

How to avoid that? Use an "a" to indicate we don't know yet when (as in target date) this will be.
```suggestion
The deprecated methods will eventually be removed in a future major version.
```